scipy/integrate/_ivp/ivp.py

```diff
@@ -451,6 +488,14 @@ def solve_ivp(fun, t_span, y0, method='RK45', t_eval=None, dense_output=False,
 
     t0, tf = float(t_span[0]), float(t_span[1])
 
+    if args is not None:
+        # Wrap the user's fun (and jac, if given) in lambdas to hide the
+        # additional parameters.
+        fun = lambda t, x, fun=fun: fun(t, x, *args)
```

Comment:
Curious - I know `event=event` is needed below, but is `fun=fun`/`jac=jac` needed here or just a precaution?

Reply:
`fun` is redefined by that statement.  If I didn't pass in `fun` like that, the original definition would be lost, and when the lambda was eventually called, it would then attempt to call itself, resulting in an error.

In the first version of the pull request, I had

        original_fun = fun
        fun = lambda t, x: original_fun(t, x, *args)

which is an alternative way to ensure that the lambda expression calls the original function.

Reply:
I noticed that `jac=jac` is not needed in the wrapper of `jac`, because the only `jac` defined is in the original `jac` in scope outside the lambda expression, so I removed it.